src/main/java/com/hivemq/security/auth/ClientToken.java

```diff
@@ -181,15 +169,15 @@ public boolean equals(final Object o) {
         if (clientId != null ? !clientId.equals(that.clientId) : that.clientId != null) return false;
         if (listener != null ? !listener.equals(that.listener) : that.listener != null) return false;
         if (username != null ? !username.equals(that.username) : that.username != null) return false;
-        if ((password.isPresent() && !that.password.isPresent()) || (!password.isPresent() && that.password.isPresent())) {
+        if (password.isPresent() ? !that.password.isPresent() : that.password.isPresent()) {
```

Comment:
so, basically that's a XOR? https://www.baeldung.com/java-xor-operator
I'd suggest to use ^ then, it clearly shows the intention that both passwords are different.